### PR TITLE
DatePicker: optional underlined prop, like for TextField

### LINKED ISSUE #5032: Feature Request: Underlined option for DatePicker
Would be nice to have an underlined option for DatePicker. Like the one TextField has.

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #5032 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

This feature was pretty easy to add, but it'd be great if design could look over it, too! It is an optional prop, so you'd only see a difference when passing `underlined` into DatePicker. Otherwise, it is rendered normally. I added a VR test as well and some screenshots below:

image

image

### CODE REVIEW

**Comment by natalieethell** [comment]
@betrue-final-final let me know what you think.

File: apps/vr-tests/src/stories/DatePicker.stories.tsx [hunk 1]
```diff
@@ -24,12 +24,18 @@ storiesOf('DatePicker', module)
...
+      {story()}
     </Screener>
   ))
   .add('Root', () => (
     <DatePicker
-      value={ date }
+      value={date}
+    />
+  ))
+  .add('Underlined', () => (
```
Comment (micahgodbolt) [inline apps/vr-tests/src/stories/DatePicker.stories.tsx]: can you add a few of the variants here? At least required

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts [hunk 2]
```diff
@@ -61,6 +61,12 @@ export interface IDatePickerProps extends IBaseProps<IDatePicker> {
    */
   ariaLabel?: string;
 
+  /**
+   * Whether or not the Textfield of the DatePicker is underlined.
+   * @default false
+   */
+  underlined?: boolean;
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts]: my only thought here is if there is value in simply exposing the Textfield props. It does hide this mode a little bit, but avoids us deprecating it in the future if we decide to expose all the Textfield props.

Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts]: Hmm interesting thought. I'm not sure if we'd ever want to expose TextField props like mask, maskChar, prefix, suffix, errorMessage, etc., but maybe I'm just overlooking a use case for those. @betrue-final-final what do you think?

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts]: nah, that's a good point. there are a TON of things in there that don't make sense to expose here. If anything, we'd want to share some lower level abstraction between datepicker and Textfield.

I'm good with this as is, and if we ever come to a situation where we need to change it, we'll do it then.

**Comment by micahgodbolt** [comment]
image

**Comment by natalieethell** [comment]
That's probably from the disabled DatePicker. I'll look into it

**Comment by micahgodbolt** [comment]
looks like 1 change incoming. Focus color of the bottom line should be primary like Textfield

image

**Comment by natalieethell** [comment]
To me, it looks like it has the same behavior as the normal DatePicker when opened. Do we want different behavior?

image

image

**Comment by micahgodbolt** [comment]
well that's kinda annoying. The non underline versions are different.

I agree that we should be consistent within component for now. Changing the datepicker to match Textfield is a bigger conversation. Happy to have the conversation, but don't want to hold up this PR to make that pretty significant change.

**Comment by micahgodbolt** [comment]
Also, the black is also how it's done currently in production

image

**Comment by betrue-final-final** [comment]
Will ok for now. FYI it's out of sync with SharePoint.
image

**Comment by micahgodbolt** [comment]
@lorejoh12 just a heads up that we're adding this variant to the datepicker. We noticed that you have your own version already, hopefully this alleviates the need for as many overrides.

As for discrepancy between outlook at SharePoint, is there design guidance there? Is moving towards black underline where your design is headed, and SharePoint should consider following?

**Pull request merged**